### PR TITLE
Introduce surgery failure chance for patients that feel pain and are awake.

### PR DESCRIPTION
**Pull request by TDSSS**
At the moment, there is no in-game reason to ever use anesthesia or allow it to be used on you for surgery, aside from RP concerns. People gleefully jump onto the table and hold relaxed conversations while bold, red messages tell them in what horrible agony they constantly are.

This PR introduces a failure chance for surgery, based on the fact that it is pretty damn hard for a patient to lie perfectly still while his skull is being sawed open.

On top of that, it should make things more interesting for antagonists that perform surgery. With the code as it is now, it is almost suspicious to insist your patient be placed under anesthesia, denying a doctor chances to for example put mindslave implants in their patients.

The exact numbers are as follows:

- Dead or unconscious patients as well as those that don't feel pain do not cause any failure chance.
- Conscious patients with hydrocodone or morphine in their bodies have a 1% failure chance and those with salicylic acid 15%.
- Ghetto anesthesia is possible with alcohol. Depending on how drunk a patient is, the failure chance can be 5%, 10% or 15%.
- Finally, without any anesthesia and no mitigating factors, the failure chance is 20%.

By now you are probably asking: But what about Vox/plasmamen/slimes, we can't anesthetize them with the tank in surgery.
There are ways beyond the anesthesia tanks to sedate a patient. You can dose them with ether or morphine in sufficient doses. You can give them hydrocodone and accept the minor failure chance. You can get them super-drunk, etc. Overall, I think the need for painkillers should have the side effect of giving the chemist some more reasons to produce more varied chemicals.

:cl:
add: Surgery failure chance for conscious patients that feel pain, can be reduced with painkillers or by getting them drunk.
tweak: Adjusts ghetto surgery tool chances up so ghetto surgery has roughly the same odds as before.
/:cl:

### CODE REVIEW

**Comment by Spartan6** [comment]
Yessssss

**Comment by Rumiluntti** [comment]
Oh fuck yes.

**Comment by Ralta** [comment]
Yeeeeees...

YEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEES.

FINALLY. MERGE IT NOW. DO IT.

File: code/modules/surgery/helpers.dm [hunk 1]
```diff
@@ -81,7 +81,26 @@
...
+		return 1
+	if(ishuman(M)) //default mob doesn't have species
+		var/mob/living/carbon/human/H = M
+		if(NO_PAIN in H.species.species_traits)//if you don't feel pain, you can hold still
+			return 1
+		if(M.reagents.has_reagent("hydrocodone"))//The most effective pain killer
+			return 0.97
+		if(M.reagents.has_reagent("morphine"))//Morphine can knock you out completely to guarantee success, but even a little bit helps
+			return 0.95
+		if(M.drunk>=80)//really damn drunk
```
Comment (SamHPurp) [inline code/modules/surgery/helpers.dm]: Fix your argument spacing please

File: code/modules/surgery/helpers.dm [hunk 2]
```diff
@@ -81,7 +81,26 @@
...
+			return 0.95
+		if(M.drunk>=40)//pretty drunk
```
Comment (SamHPurp) [inline code/modules/surgery/helpers.dm]: Fix your argument spacing please

File: code/modules/surgery/helpers.dm [hunk 3]
```diff
@@ -81,7 +81,26 @@
...
+			return 0.9
+		if(M.reagents.has_reagent("sal_acid")) //it's better than nothing, as far as painkillers go.
+			return 0.85
+		if(M.drunk>=15)//a little drunk
```
Comment (SamHPurp) [inline code/modules/surgery/helpers.dm]: Fix your argument spacing please

**Comment by granodd** [comment]
A PR that makes pain in surgery actually mean something? Yes, please. Maybe this will stop people who just rush in and say "put this in me: no anesthesia" and then just climb onto the table with little grace afterwards.

File: code/modules/surgery/helpers.dm (hunk 1 above)
Comment (TDSSS) [reply code/modules/surgery/helpers.dm]: Thanks. Fixed them all (hopefully)

**Comment by FalseIncarnate** [comment]
People also should remember that being asleep is considered unconscious so things like ether are highly useful for your non-breathing patients.

You can also ask atmos to whip up a nice tank of vox-safe anesthetics, any atmos tech should (hopefully) know how to mix gases.

**Comment by EldritchSigma** [comment]
Pretty nice. Finally no 3dgy "I dont feel pain." Bullshit. It even says that your skin feels like its on fire. Of course you feel pain.

**Comment by GunDOSMk1** [comment]
Great idea, I give my thumbs up to this!

File: code/modules/surgery/surgery.dm [hunk 4]
```diff
@@ -115,6 +115,8 @@
 		if(implement_type)	//this means it isn't a require nd or any item step.
 			prob_chance = allowed_tools[implement_type]
 		prob_chance *= get_location_modifier(target)
+		prob_chance *= get_pain_modifier(target)//operating on conscious people is hard.
```
Comment (Crazylemon64) [inline code/modules/surgery/surgery.dm]: What if the step does not cause pain, like repairing/attaching a bionic limb?

**Comment by DesolateG** [comment]
If hydrocodone and morphine mute pain, why would their be a failure chance?

File: code/modules/surgery/helpers.dm [hunk 5]
```diff
@@ -81,7 +81,26 @@
 			return 1
 	return 0
 
-
+/proc/get_pain_modifier(mob/M) //returns modfier to make surgery harder if patient is conscious and feels pain
```
Comment (Citinited) [inline code/modules/surgery/helpers.dm]: Can you typecast this to `mob/living/carbon/human`?

**Comment by FalseIncarnate** [comment]
@DesolateG probably because the person is still conscious and would be able to flinch, as that reaction is not tied to the pain response but rather the anticipation/expectation of pain. Knocking them out completely would negate their ability to anticipate the action and thus would suppress flinching.

File: code/modules/surgery/surgery.dm (hunk 4 above)
Comment (TDSSS) [reply code/modules/surgery/surgery.dm]: Good question, I hadn't considered that. Though attaching a bionic limb probably should be painful, since I assume you'd need to attach the nerves to the new limb and so on.

I agree on the repairing, though. Will look into this.

**Comment by ktoma36** [comment]
The only problem I see with this PR is that with xeno infestations progressing as fast as they do now every second counts. Of course the chance of xeno's showing up is pretty rare but it should be taken into consideration. Other than that 👍

File: code/modules/surgery/helpers.dm (hunk 5 above)
Comment (TDSSS) [reply code/modules/surgery/helpers.dm]: I could, yeah. It gets typecast in the proc itself as is. I tried to make it the same as the other helper proc that gets called in surgery, 'get_location_modifier(mob/M)'.

I don't think there should be a big difference between doing it here or in the proc, so it's mostly a style thing. What do others think?

**Comment by TDSSS** [comment]
@DesolateG Part of the reason is that it only mutes pain, it doesn't eliminate it. And as FalseIncarnate says, you probably still move around, flinch and twitch at the sounds and so on.

On a more meta level, injecting some morphine is a lot easier and less of a hassle, so it should come with some minor downsides to compensate. It's not like a 3% failure chance is very crippling.

**Comment by DesolateG** [comment]
@FalseIncarnate  bad example... I was fully awake when my wisdom teeth were removed. I've also had multiple surgeries where I was awake but numb to the pain.

**Comment by FalseIncarnate** [comment]
I never suggested being awake is going to make surgery impossible, if that's how you interpreted my statements then it appears things were unclear. What I was suggesting is that someone awake is far more likely to flinch, which could disrupt certain steps. You may have been able to sit through a surgery without flinching, but (and this should not surprise anyone) not everyone is you. Numbness is going to definitely reduce the chances of them moving, but not fully prevent it... which is what those small fail chances are. You are far more likely to succeed with painkillers alone than without any aid, but you're going to have the absolute best chance if you can knock them out completely so they quit moving altogether.

**Comment by alexkar598** [comment]
I don't think it's a great idea,in some cases you really need to treat it fast

Alien surgery
Internal bleeding
Punctured lungs with no salbutamol in patient

These are all cases where giving anaesthetics is not a option
Also you can always just use the lovely sleep command which would begate this WHOLE pr

**Comment by alexkar598** [comment]
Also injecting morphine is mostly a bad idea,it has a high addiction chance and it will create you an other problem


As a medical player I can say that we can get very busy and we don't need an other problem

**Comment by TDSSS** [comment]
It's pretty easy to include a check for sleeping.

And if you don't want to use morphine, use hydrocodone. It literally takes only a handful of seconds to inject someone with a syringe, if you have it prepared.

**Comment by alexkar598** [comment]
That would neglect ether ability to be anesthetic making it useless for the whole medbay might as well  remove ether at all

File: code/modules/surgery/surgery.dm [hunk 6]
```diff
@@ -116,6 +116,10 @@
 			prob_chance = allowed_tools[implement_type]
 		prob_chance *= get_location_modifier(target)
 
+		if(!istype(surgery.steps[surgery.status], /datum/surgery_step/robotics))//Repairing robotic limbs doesn't hurt
```
Comment (TDSSS) [inline code/modules/surgery/surgery.dm]: So, this currently doesn't actually work as intended, for some reason. It's probably something really obvious I'm overlooking.

**Comment by TDSSS** [comment]
By my reading of the code, ether uses Paralyse() to set the paralysis var, which is different from the sleeping var.

**Comment by Gummin** [comment]
Yes please x :100:

**Comment by scrubmcnoob** [comment]
but muh hugbox
On a serious note
Few questions.
What's the penalty for failing a surgery exactly? What will stop me from just spamming a certain step till I succeed and carry on with the surgery? Will it damage the patient/make them lose more blood?-
Another is, can weed be added as a ghetto way of anesthesia, not a powerful one to any degree but just enough so we got a valid excuse to smoke a fat doobie during surgery?

At any rate, this looks really cool and hope to see it merged in. Enforcing RP via mechanics is great.

**Comment by DesolateG** [comment]
@scrubmcnoob  Do ghetto surgery some time. It causes brute damage if it fails.

**Comment by Varghh** [comment]
I like the idea, but add Morphine and Hydrocodone to the mix, if this means i have the use the tank EVERY time it's going to be a pain when dealing with terror-xenos-slings to the points is going to drive people insane.

**Comment by pinatacolada** [comment]
Ehhhhhhhhh idk how i feel about punishing players for ysing hydrocodone. Its not even a round start chem.

Otherwise id even consider upping the fail chance when not using any sedatives, like up to 30 maybe 35.

**Comment by Denthamos** [comment]
I like this, would be nice to finally see anesthetics be important in the surgery process.  It makes sense a patient not sedated would jolt and shuffle around when large amounts of pain is experienced or certain nerve endings are stimulated.

**Comment by KasparoVy** [comment]
Thank you very much!

**Comment by Hylocereus** [comment]
YESSSSSS I love this! As someone who almost exclusively plays medical, I can tell you that the "BUT IT TAKES TOO LONG REE" crap is just that - crap. It takes less than 5 seconds to shoot someone up with morphine if you're in a hurry. 

If they die in those five seconds tough shit, that's what happens when you don't stabilize your patient before you operate. It's not hard. No time to stabilize? Sometimes things go bad. It happens. Learn to deal with it instead of insisting everything be made easier. If an officer waits five minutes before going to medbay with a xeno larva in them that's their fault. If someone is brought in with 40% blood and internal bleeding hook up the IV and slap some salb/Perflu in them before you operate. Even an emergency autoinjector is enough. The internal bleeding killing people through surgery thing is very overblown, it's not difficult to prevent and nothing stops you from immediately defibbing with the 'fibs in the surgery locker anyway.

File: code/modules/surgery/helpers.dm (hunk 5 above)
Comment (SamHPurp) [reply code/modules/surgery/helpers.dm]: I agree with @Citinited

**Comment by taukausanake** [comment]
If people are still complaining about needing to put someone to sleep because they are too lazy to add in an extra step, how about making is so you can restrain someone? Tie them down and perform surgery? Might be faster than putting someone to sleep and more morbid too. I'd do it just to cut off heads for Robotics

File: code/modules/surgery/helpers.dm (hunk 5 above)
Comment (TDSSS) [reply code/modules/surgery/helpers.dm]: Okay, changed it.

**Comment by Balthusdire** [comment]
I am 1000% for this. I like how the fail chance is low enough that it isn't debilitating to do it without anesthetic if needed, but should still force it to be used in normal circumstances. This is what I have been hoping to see for a long time.

**Comment by Regen1** [comment]
This is great, I've wanted this for a long time.

**Comment by TullyBurnalot** [comment]
While I love the idea of making anesthesia way more needed for more than just RP reasons, the balance effects of this are particularly severe. Especially for Ghetto surgery.  I'd like to see the failure chances of ghetto surgery reduced to compensate for this, so that ghetto surgery is able to be done with roughly the same failure rates as before. Additionally, morphine and hydrocodone should allow a near 100% chance for success. Having a failure rate on those would be a nerf to those species who can't use N2O. 

Tl;dr - Love the concept in theory, in practice the numbers will need some tweaking.

**Comment by alexkar598** [comment]
Well it's time to prepare those neurotoxin injections

On 2017-09-28, at 10:57 AM, "TullyBurnalot" <notifications@github.com<mailto:notifications@github.com>> wrote:


While I love the idea of making anesthesia way more needed for more than just RP reasons, the balance effects of this are particularly severe. Especially for Ghetto surgery. I'd like to see the failure chances of ghetto surgery reduced to compensate for this, so that ghetto surgery is able to be done with roughly the same failure rates as before. Additionally, morphine and hydrocodone should allow a near 100% chance for success. Having a failure rate on those would be a nerf to those species who can't use N2O.

Tl;dr - Love the concept in theory, in practice the numbers will need some tweaking.

**Comment by TDSSS** [comment]
@TullyBurnalot 
Let me address your points separately. 

> While we love the idea of making anesthesia way more needed for more than just RP reasons

Who exactly is 'we'?

> We'd like to see the failure chances of ghetto surgery reduced to compensate for this, so that ghetto surgery is able to be done with roughly the same failure rates as before. 

Something I've thought about as well, probably worth implementing. It would mean messing up the usually 'neat' numbers on all the tools and going through all the surgeries. I'll take a look.

>Additionally, morphine and hydrocodone should allow a near 100% chance for success. 

That was already my intention. 95 and 97 seemed close enough to 100 for that purpose to me. I suppose I could increase it to something like 99%, maybe. Though it seems a bit overkill.

>Having a failure rate on those would be a nerf to those species who can't use N2O.

The only species that can't use N2O are Slimes who don't breath. There's nothing preventing a Vox or Plasmaman from using an N2O+Plasma/N2 mix for anesthesia, or from just getting some Ether instead.

**Comment by TullyBurnalot** [comment]
> Who exactly is 'we'?

That'd uh, be the Heads of Staff and Maintainers. Apologies, should've made it clearer.

@TDSSS

**Comment by TDSSS** [comment]
Alright, I adjusted the failure chance of morphine and Hydrocodone and tweaked ghetto surgery chances to give roughly the same odds as before.

**Comment by SamHPurp** [comment]
These are good changes, I hadn't considered the ghetto implications. 👍

**Comment by Balthusdire** [comment]
What about borgs. Medical borgs have no way to administer any sort of painkiller or anesthetic and I would argue that is a big issue. They need some reliable way to deal with this new failure rate if it goes through.

**Comment by taukausanake** [comment]
@Balthusdire I think #8146 solved that?

**Comment by Balthusdire** [comment]
@taukausanake Yes it does, I did not see that. Excellent!

**Comment by Dualzed** [comment]
Honestly, this idea is great, the problem is that, for the experienced players, they do a surgery in 15 seconds. There is also robotics to think about, in reality this would be super amazing, on a high rp server, currently rp is not enforced at all, and people are there to play the game, they want to go in out of medbay as soon as possible.  
  
There is also the fact how it will mess up ghetto, and the fix for that will just be messy,  
  
I love the idea, but not on a low rp server where powergaming runs rampant.

**Comment by TDSSS** [comment]
What do you mean with 'there is also robotics to think about'? Robotics starts with an anesthesia tank like the ORs do, and repairing robotic limbs doesn't cause pain.

And well...I already did the fix for ghetto surgery, it was hardly messy to change some numbers.

**Comment by ktoma36** [comment]
What about people who have augmented body parts such as chest and heads? Coming across someone needing surgery with such augments would be rare of course but it does happen. While it could just be easily chucked up under 'you knew what you were getting into' when you get augmented I am curious as to if augmented body parts could cause issues. Augmented body parts can cause pain, if it wasnt already common knowledge.

**Comment by TDSSS** [comment]
>What about people who have augmented body parts such as chest and heads?

All 'robot surgery' (As in, using a screwdriver to unscrew hatches and so on) doesn't cause a pain failure chance.

**Comment by ktoma36** [comment]
Augmented chest require both a screwdriver and a scalpel I believe (thats what I have been told)

**Comment by TDSSS** [comment]
Yeah, you need a scalpel to actually start/select the surgery, it brings up the 'select surgery' menu. But all actual surgery steps use screwdrivers, IIRC.

**Comment by ktoma36** [comment]
What about when you start working with organic organs in the body? The only reason Im really bringing this up is to save headaches later

**Comment by TDSSS** [comment]
Okay, after some testing manipulating organic organs in an augmented chest uses normal IPC surgery steps, IE you use multitools to remove organs, nanopaste to repair them and so on.

So no issues there.

**Comment by EvadableMoxie** [comment]
I have no problem with the idea of forcing anesthesia in general.

My only concern here is that it makes ghetto surgery incredibly difficult.  The failure chances are already quite high, adding another stacking failure chance on top of it will really make it even more of a niche thing.

I think ghetto surgery is long overdo for an overhaul anyway, though.

**Comment by TDSSS** [comment]
So, I haven't heard anything for a while from the Heads/maintainers. Is there anything else that should change? Or could this be merged soon?

Sorry if I sound impatient, I have no idea how long those things usually take.

**Comment by taukausanake** [comment]
Don't worry @TDSSS this is kind of normal. Just be patient

**Comment by tigercat2000** [comment]
Manually merged in commit 228295b. Learning experience for everyone: GitHub does not automatically mark a PR as merged unless you merge it _exactly as specified in the directions_ and do not do any rebasing.

**Comment by TDSSS** [comment]
Hah, weird. I assume I can still delete this branch without issues?

**Comment by tigercat2000** [comment]
@TDSSS Yeah, you're good.
